packages/opentelemetry-metrics/src/Metric.ts

```diff
@@ -77,6 +88,33 @@ export abstract class Metric<T> implements types.Metric<T> {
     return;
   }
 
+  /**
+   * Provides a ReadableMetric with one or more TimeSeries.
+   * @returns The ReadableMetric, or null if TimeSeries is not present in
+   *     Metric.
+   */
+  get(): ReadableMetric | null {
+    if (this._handles.size === 0) return null;
+
+    const timestamp = hrTime();
+    return {
+      descriptor: this._metricDescriptor,
+      timeseries: Array.from(this._handles, ([_, handle]) =>
+        ((handle as unknown) as BaseHandle).getTimeSeries(timestamp)
+      ),
+    };
+  }
+
+  private _getMetricDescriptor(): MetricDescriptor {
+    return {
+      name: this._name,
+      description: this._options.description,
+      unit: this._options.unit,
+      labelKeys: this._options.labelKeys,
+      type: MetricDescriptorType.GAUGE_INT64,
```

Comment:
Decide which `MetricDescriptorType` to assign in case of Gauge (two options are `GAUGE_INT64` and `GAUGE_DOUBLE`) and in case of counter (two options are `COUNTER_INT64` and `COUNTER_DOUBLE`)?

Reply:
Double can represent integers and decimals and looks like a safe default, doesn't it?
Another way would be to infer it from the data that is in that metric?

Reply:
> Double can represent integers and decimals and looks like a safe default, doesn't it?

For now, I have updated the PR to use `GAUGE_DOUBLE` and `COUNTER_DOUBLE` MetricDescriptorType for `Gauge` and `Counter` Metrics respectively.

> Another way would be to infer it from the data that is in that metric?

AFAIK `MetricDescriptorType` needs to be defined at the time of metric creation and recorded value should honor the type and not the vice-versa.